stdlib/base.pkl

```diff
@@ -1367,6 +1367,12 @@ external class String extends Any {
   /// Splits this string around matches of [pattern].
   external function split(pattern: String|Regex): List<String>
 
+  /// Splits this string matches of [pattern], up to [limit] substrings.
```

Comment:
What if you use 0 or -1 as a limit? 🤔
Would it return nothing or a full list?

I guess it makes sense to document that behavior.

Reply:
Unlike Java, we can define `limit` to _not_ include 0 or negative numbers! Perhaps this should be `limit: Int(this > 0)`

Reply:
I think it might just be best (and simplest) to mimic (and document) the Java behavior here, which is 

> * If the limit `n` is greater than zero then the pattern will be applied at most `n - 1` times, the array's length will be no greater than `n`, and the array's last entry will contain all input beyond the last matched delimiter. > * If `n` is non-positive then the pattern will be applied as many times as possible and the array can have any length.
> * If `n` is zero then the pattern will be applied as many times as possible, the array can have any length, and trailing empty strings will be discarded.

Reply:
> Unlike Java, we can define `limit` to _not_ include 0 or negative numbers! Perhaps this should be `limit: Int(this > 0)`

The question is, if pkl needs that behvior ? 🤔 
Are there use cases for -1 and 0 (as described by @HT154)? 🤔 

Additionally, instead of making this function super complicated to understand (with these rules) we might want to introduce other functions for the "zero and negative number" behavior? 
Not sure if this is a good idea though...

Reply:
`splitLimit(char, -1)` is the same as `split(char)`

`splitLimit(char, 0)` is, weirdly, `split(char).dropLastWhile((it) -> it.empty)`, which is weirdly specific (and strange).

In both of these cases, I don't really see a strong use-case for them. 

Side-note: many of our stdlib APIs expect positive values, for example, `List.drop()` and `List.take()`, but this check is implemented in the external method rather than using `UInt` as a type. We should probably fix them.

Reply:
I agree with you.
And since pkl is a configuration language (and not a full programming language), I do think it makes sense to add `limit > 0` ✅

This will remove a lot of complexity for things that are rarely (if even) used.

Reply:
Updated. Is it also worth checking this in the external code?

Reply:
It's not necessary to check in the external code, if we ever get a limit <= 0 there it's a bug we should fix.